pkg/render/fluentd_test.go

```diff
@@ -213,11 +214,148 @@ var _ = Describe("Tigera Secure Fluentd rendering tests", func() {
 
 	})
 
+	It("should render with splunk configuration with ca", func() {
+		splkCreds := &render.SplunkCredential{
+			Token:       []byte("TokenForHEC"),
+			Certificate: []byte("Certificates"),
+		}
+		instance.Spec.AdditionalStores = &operatorv1.AdditionalLogStoreSpec{
+			Splunk: &operatorv1.SplunkStoreSpec{
+				Endpoint:    "https://1.2.3.4:8088",
+			},
+		}
+		component := render.Fluentd(instance, nil, esConfigMap, s3Creds, splkCreds, filters, eksConfig, nil, installation)
+		resources, _ := component.Objects()
+		Expect(len(resources)).To(Equal(4))
+
+		// Should render the correct resources.
+		expectedResources := []struct {
+			name    string
+			ns      string
+			group   string
+			version string
+			kind    string
+		}{
+			{name: "tigera-fluentd", ns: "", group: "", version: "v1", kind: "Namespace"},
+			{name: "logcollector-splunk-credentials", ns: "tigera-fluentd", group: "", version: "v1", kind: "Secret"},
+			{name: "logcollector-splunk-public-certificate", ns: "tigera-fluentd", group: "", version: "v1", kind: "Secret"},
+			{name: "fluentd-node", ns: "tigera-fluentd", group: "apps", version: "v1", kind: "DaemonSet"},
+		}
+
+		i := 0
+		for _, expectedRes := range expectedResources {
+			ExpectResource(resources[i], expectedRes.name, expectedRes.ns, expectedRes.group, expectedRes.version, expectedRes.kind)
+			i++
+		}
+
+		ds := resources[3].(*apps.DaemonSet)
```

Comment:
Since this test is rendering with CA, I think it'd be good to check the volumes/volumeMounts.

Reply:
added a generic check for volumes length in both test cases, explicate check with volume name in CA case.